spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-nacos-discovery/src/main/java/com/alibaba/cloud/nacos/NacosDiscoveryProperties.java

```diff
@@ -159,6 +159,11 @@ public class NacosDiscoveryProperties {
 	 */
 	private String ip;
 
+	/**
+	 * if nacos using ip address for service, default is true.
+	 */
+	private boolean preferIpAddress = true;
```

Comment:
I think that adding a new field called hostname for holding hostname is better.
and we define a rule below:
1.If user sets IP address, IP address preferred.
2.If user doesn't set IP address and set hostname together, we extract IP address from hostname.

Reply:
I think, the property ip should changed to host is better,  host can be ip or hostname.  and the nacos client's instance just with property  'ip',  not host.  Can this change to host?